### ISSUE TITLE
let's use `Expect` in all atoms

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#3461]
Currently, we use `Expect` class just in a few atoms, where we want to have a detailed error reporting in case of user errors. Let's use it in _all_ atoms.

**Comment by asmirnov-backend** [comment]
@yegor256 Should `Expect` be used for checking `Attr.RHO`?
For examlpe, transform this 
```java
new Dataized(this.take(Attr.RHO)).asNumber()
```
to this
```java
Expect.at(this, Attr.RHO)
      .that(phi -> new Dataized(phi).asNumber())
      .otherwise("Attr.RHO must be a number for 'EOnumber'")
      .it();
```

**Comment by yegor256** [comment]
@asmirnov-backend yes, definitely, this will be helpful

**Comment by asmirnov-backend** [comment]
While using `Expect` some code duplicates appear, for example, next piece of code used in `EOnumber$EOdiv`, `EOnumber$EOgt`, `EOnumber$EOplus`, `EOnumber$EOtimes` (Ref: #3789)
```java
Expect.at(this, Attr.RHO)
    .that(phi -> new Dataized(phi).asNumber())
    .otherwise("must be a number")
    .it();
```
Perhaps it would be useful to have a more clever `Dataized` with `Expect` class.
```java
public class DataizableExpect {

    private final Expect<Phi> expect;

    public DataizableExpect(final Expect<Phi> expect) {
        this.expect = expect;
    }

    public Double asDouble() {
        return this.expect
            .that(phi -> new Dataized(phi).asNumber())
            .otherwise("must be a number")
            .it();
    }

    public Integer asPositiveInt() {
        return this.expect
            .that(phi -> new Dataized(phi).asNumber())
            .otherwise("must be a number")
            .must(number -> number % 1 == 0)
            .that(Double::intValue)
            .otherwise("must be an integer")
            .must(integer -> integer >= 0)
            .otherwise("must be a positive integer")
            .it();
    }

}
```
The code that uses this class would look something like this:
```java
new DataizableExpect(Expect.at(this, Attr.RHO)).asDouble()
```
@yegor256 What do you think?

**Comment by yegor256** [comment]
@asmirnov-backend maybe we can make inner classes:

```
new Expect.Number(Expect.at(this, Attr.RHO)).it();
new Expect.Integer(Expect.at(this, Attr.RHO)).it();
new Expect.Whole(Expect.at(this, Attr.RHO)).it(); // greater or equal to zero
```

WDYT?

**Comment by asmirnov-backend** [comment]
@yegor256 Nice idea, thanks

**Comment by sherenkov** [comment]
@yegor256 @maxonfjvipon — is this issue still active and open for outside contributors? It's tagged `help wanted` and `good first issue`, but the signal from recent activity is mixed:

- The most recent attempt, #3853 (`Expect` for `EOmalloc$EOof$EOallocated$EOread`), was closed silently in April 2026 without review or comment.
- The last design discussion here is from January 2025.
- Many atoms still don't use `Expect` — e.g. `EOi64$EOdiv|gt|plus|times`, the `EObytes$*` bitwise ops, and the rest of the `EOmalloc$*` cluster.

If the issue is still wanted, I'd like to pick the smallest natural cluster to start: the four `EOi64` arithmetic atoms (`div`, `gt`, `plus`, `times`), mirroring the pattern already merged in #3789 for the corresponding `EOnumber` atoms. That would keep the diff well under the size limit and leave the issue open for the remaining clusters in follow-up PRs.

If you agree, please assign this issue to me. Otherwise, please let me know if the project's appetite for this work has shifted so I can pick another ticket.

**Comment by sherenkov** [comment]
@yegor256 follow-up to merged #5117: PR #5131 is ready for review. All 26 checks pass.

It migrates the three remaining numeric **converter** atoms in this family to `Expect`:

- `EOi16$EOas_i32` → new `Expect.I16`
- `EOi32$EOas_i64` → new `Expect.I32`
- `EOi64$EOas_number` → reuses `Expect.I64`

Diff is +123/−3 across 5 files. No PMD suppressions (per your #5117 feedback).

Note on scope: after #5117 landed, the only atoms left in the `i16`/`i32`/`i64` family that still bypass `Expect` are these converters — `i32.div|gt|plus|times` etc. are EO objects in `i32.eo` that delegate to `i64` via `as-i64`, not Java atoms.

Issue stays open for the remaining clusters (`EObytes$*` bitwise ops, `EOmalloc$*`).

**Comment by sherenkov** [comment]
@yegor256 here's status on this:

  Merged: #5117, #5131, #5138, #5146 — every atom with a numeric precondition is now covered. 

  Opened PR #5147 (write/φ) is the last numeric one.

  The remaining atoms have no precondition Expect can assert:

  - bytes atoms (and/concat/eq/not/or/size/xor) — input is dataized to byte[]; in EO every Φ has a bytes representation, so the check is a no-op.
  - error — message is an opaque payload; any Φ is valid by design.
  - try — main/catch/finally are computations, and EO has no callable subtype to assert against.

**Comment by yegor256** [comment]
@sherenkov thanks for your contribution!
